src/main/scala/zio/telemetry/Telemetry.scala

```diff
@@ -0,0 +1,213 @@
+package zio.telemetry
+
+import io.opentracing.Span
+import io.opentracing.SpanContext
+import io.opentracing.Tracer
+import io.opentracing.propagation.Format
+import io.opentracing.tag.Tag
+import java.util.concurrent.TimeUnit
+import scala.jdk.CollectionConverters._
+import zio.Exit
+import zio.FiberRef
+import zio.IO
+import zio.Task
+import zio.UIO
+import zio.ZIO
+import zio.ZManaged
+import zio.clock.Clock
+
+trait Telemetry extends Serializable {
+  def telemetry: Telemetry.Service
+}
+
+object Telemetry {
+
+  trait Service {
+    def spanRef: FiberRef[Span]
+    def tracer: Tracer
+  }
+
+  def managedService(
+      tracer: Tracer,
+      rootOpName: String = "ROOT"
+  ): ZManaged[Clock, Nothing, Telemetry.Service] =
+    ZManaged.make(
+      for {
+        span <- UIO(tracer.buildSpan(rootOpName).start())
+        ref <- FiberRef.make(span)
+        tracer_ = tracer
```

Comment:
Is this one needed?

Reply:
Without it, line 37 is compiled into a recursive call.